aries_cloudagent/connections/base_manager.py

```diff
@@ -170,24 +174,31 @@ async def create_did_document(
                 svc_endpoint,
             )
             did_doc.set(service)
-
         return did_doc
 
-    async def store_did_document(self, did_doc: DIDDoc):
+    async def store_did_document(self, did_doc: DIDDocument):
+        return await self._store_did_document(did_doc,did_doc.id)
+
+    
+    async def store_did_document_with_different_id(self, did_doc: DIDDocument, unqualified_did:str):
+        """TO HANDLE LEGACY CONNECTION UNQUALIFIED DIDS WITH NEW DIDDocuments """
+        return await self._store_did_document(did_doc,unqualified_did)
+
+    async def _store_did_document(self, did_doc: DIDDocument, storage_id: Union[str,DID]):
```

Comment:
How do we feel about this? @swcurran @dbluhm It allows us to get rid of the old DIDDoc class and always use the pydid DIDDocument classes for diddocuments while allowing old unqualified dids for now.

Reply:
Works for me, although I’m perhaps not the best judge. We want to use pydid as much as possible (completely). As the issues with Peer DID get sorted out, we can evolve usage to get things cleaner.